### PR TITLE
expression: implement vectorized evaluation for builtinConvertTzSig

### LINKED ISSUE #12101: Call For Participation: vectorize all Date/Time builtin functions
This issue is used to trace all vectorized Data/Time builtin functions, which is a part of https://github.com/pingcap/tidb/issues/12058.

**NOTE**:
1. Feel free to join us in the [`wg-vec-expr` working group](https://github.com/pingcap/community/blob/master/working-groups/wg-vec-expr.md)
2. Please comment on this issue if you are working on any of the functions.

**Note For Performance Challenge Teams: each PR on this issue has 50 score.**

-----
- [x] builtinCastIntAsTimeSig
- [x] builtinCastRealAsTimeSig
- [x] builtinCastDecimalAsTimeSig
- [x] builtinCastStringAsTimeSig
- [x] builtinCastTimeAsTimeSig
- [x] builtinCastDurationAsTimeSig
- [x] builtinCastJSONAsTimeSig
- [x] builtinCoalesceTimeSig
- [x] builtinCaseWhenTimeSig
- [x] builtinIfTimeSig  #12140
- [x] builtinIfNullTimeSig
- [x] builtinTimeAnyValueSig
- [x] builtinNameConstTimeSig
- [x] builtinDateSig @hey-kong https://github.com/pingcap/tidb/pull/12136
- [x] builtinDateLiteralSig
- [x] builtinFromDaysSig
- [x] builtinFromUnixTime1ArgSig
- [x] builtinStrToDateDateSig
- [x] builtinStrToDateDatetimeSig
- [x] builtinSysDateWithFspSig
- [x] builtinSysDateWithoutFspSig
- [x] builtinCurrentDateSig
- [x] builtinUTCDateSig
- [x] builtinUTCTimestampWithArgSig
- [x] builtinUTCTimestampWithoutArgSig
- [x] builtinNowWithArgSig
- [x] builtinNowWithoutArgSig
- [x] builtinTimestamp1ArgSig @Reminiscent https://github.com/pingcap/tidb/pull/12339
- [x] builtinTimestamp2ArgsSig @Reminiscent https://github.com/pingcap/tidb/pull/12352
- [x] builtinTimestampLiteralSig
- [x] builtinAddTimeDateTimeNullSig
- [x] builtinAddDatetimeAndDurationSig
- [x] builtinAddDatetimeAndStringSig
- [ ] builtinConvertTzSig
- [x] builtinMakeDateSig
- [x] builtinLastDaySig
- [x] builtinTidbParseTsoSig
---

Prefer to implement them by go generate, see https://docs.google.com/document/d/17XdYo_EwYaWANqbjS3i3OVSz3oLlovU2fIo1bRx4YqA/edit#, and find your interest and take a job.

builtinAddDateStringStringSig
builtinAddDateStringIntSig
builtinAddDateStringRealSig
builtinAddDateStringDecimalSig
builtinAddDateIntStringSig
builtinAddDateIntIntSig
builtinAddDateIntRealSig
builtinAddDateIntDecimalSig
builtinAddDateDatetimeStringSig
builtinAddDateDatetimeIntSig
builtinAddDateDatetimeRealSig
builtinAddDateDatetimeDecimalSig
builtinAddDateDurationStringSig
builtinAddDateDurationIntSig
builtinAddDateDurationRealSig
builtinAddDateDurationDecimalSig

builtinSubDateStringStringSig
builtinSubDateStringIntSig
builtinSubDateStringRealSig
builtinSubDateStringDecimalSig
builtinSubDateIntStringSig
builtinSubDateIntIntSig
builtinSubDateIntRealSig
builtinSubDateIntDecimalSig
builtinSubDateDatetimeStringSig
builtinSubDateDatetimeIntSig
builtinSubDateDatetimeRealSig
builtinSubDateDatetimeDecimalSig
builtinSubDateDurationStringSig
builtinSubDateDurationIntSig
builtinSubDateDurationRealSig
builtinSubDateDurationDecimalSig

builtinSubDateAndDurationSig
builtinSubDateAndStringSig
builtinSubDatetimeAndDurationSig
builtinSubDatetimeAndStringSig
builtinSubDurationAndDurationSig
builtinSubDurationAndStringSig
builtinSubStringAndDurationSig
builtinSubStringAndStringSig
builtinSubTimeDateTimeNullSig
builtinSubTimeDurationNullSig
builtinSubTimeStringNullSig

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: close #12101 

Problem Summary: implement vectorized evaluation for builtinConvertTzSig

### Check List 

Tests 

- Unit test

### CODE REVIEW

File: expression/builtin_time_vec.go [hunk 1]
```diff
@@ -2519,11 +2519,50 @@ func (b *builtinUTCTimestampWithoutArgSig) vecEvalTime(input *chunk.Chunk, resul
...
+
+	result.MergeNulls(fromTzBuf, toTzBuf)
+	ts := result.Times()
+	var isNull bool
+	for i := 0; i < n; i++ {
+		if result.IsNull(i) {
+			continue
+		}
+
+		ts[i], isNull, err = b.convertTz(ts[i], fromTzBuf.GetString(i), toTzBuf.GetString(i))
```
Comment (Reminiscent) [inline expression/builtin_time_vec.go:2557]: When the `fromTzBuf.GetString(i)` or `toTzBuf.GetString(i)` is Null. We should directly get the `types.ZeroTime`. See [here](https://github.com/pingcap/tidb/blob/master/expression/builtin_time.go#L5503-L5511).

Comment (Reminiscent) [reply expression/builtin_time_vec.go:2557]: Besides, please add tests to cover this case.

Comment (qw4990) [reply expression/builtin_time_vec.go:2557]: My mistake. It is fixed https://github.com/pingcap/tidb/pull/16349/commits/bf8bdc634e467e8c7a6a1b10a01b64010507c160

**Comment by qw4990** [comment]
PTAL. This is the last one that needs to be vectorized. @XuHuaiyu @Reminiscent @wshwsh12

**Pull request merged**